integration-tests/tests/snap_op_test.go

```diff
@@ -49,3 +55,20 @@ func (s *snapOpSuite) testInstallRemove(c *check.C, snapName, displayName string
 func (s *snapOpSuite) TestInstallRemoveAliasWorks(c *check.C) {
 	s.testInstallRemove(c, "hello-world", "hello-world")
 }
+
+func (s *snapOpSuite) TestRemoveRemovesAllRevisions(c *check.C) {
+	snapPath, err := build.LocalSnap(c, data.BasicSnapName)
+	defer os.Remove(snapPath)
+	c.Assert(err, check.IsNil, check.Commentf("Error building local snap: %s", err))
+
+	// install two revisions
+	installSnap(c, snapPath)
+	installOutput := installSnap(c, snapPath)
+	c.Assert(installOutput, testutil.Contains, data.BasicSnapName)
+	// double check
+	revnos, _ := filepath.Glob(filepath.Join(dirs.SnapSnapsDir, data.BasicSnapName, "1*"))
```

Comment:
Maybe a small comment that the glob is on `1*` because we want to exclude the "current" symlink (and that all sideloaded ones start with 10000 or something)?